File: api/characters.py
from http.server import BaseHTTPRequestHandler
import json

# 静的なキャラクター定義（高速化のため）
CHARACTERS = {
    "reimu": {"name": "博麗霊夢", "full_name": "博麗 霊夢（はくれい れいむ）", "description": "博麗神社の巫女", "avatar": "/avatars/reimu.png"},
    "marisa": {"name": "霧雨魔理沙", "full_name": "霧雨 魔理沙（きりさめ まりさ）", "description": "普通の魔法使い", "avatar": "/avatars/marisa.png"},
    "sakuya": {"name": "十六夜咲夜", "full_name": "十六夜 咲夜（いざよい さくや）", "description": "紅魔館のメイド長", "avatar": "/avatars/sakuya.png"},
    "yuyuko": {"name": "西行寺幽々子", "full_name": "西行寺 幽々子（さいぎょうじ ゆゆこ）", "description": "白玉楼の亡霊", "avatar": "/avatars/yuyuko.png"}
}

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Cache-Control', 'max-age=3600')  # 1時間キャッシュ
        self.end_headers()
        
        # 高速化：事前に構築されたレスポンス
        character_list = [
            {"id": char_id, **char_data}
            for char_id, char_data in CHARACTERS.items()
        ]
        
        response = {"characters": character_list}
        self.wfile.write(json.dumps(response, ensure_ascii=False).encode('utf-8'))
        
    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()

File: app/constants/characters.ts
// キャラクター情報の型定義
export interface Character {
  id: string;
  name: string;
  full_name: string;
  description: string;
  avatar: string;
}

// 静的キャラクターデータ - 即座読み込み用
export const CHARACTERS: Character[] = [
  {
    id: 'reimu',
    name: '博麗霊夢',
    full_name: '博麗 霊夢（はくれい れいむ）',
    description: '楽園の巫女。いつも気怠そうにしているが、実は強い責任感を持つ。',
    avatar: '/characters/reimu.png'
  },
  {
    id: 'marisa',
    name: '霧雨魔理沙',
    full_name: '霧雨 魔理沙（きりさめ まりさ）',
    description: '普通の魔法使い。努力家で向上心が強く、常に新しい魔法を研究している。',
    avatar: '/characters/marisa.png'
  },
  {
    id: 'sakuya',
    name: '十六夜咲夜',
    full_name: '十六夜 咲夜（いざよい さくや）',
    description: '紅魔館のメイド長。時を操る能力を持ち、完璧な仕事ぶりで知られる。',
    avatar: '/characters/sakuya.png'
  },
  {
    id: 'yuyuko',
    name: '西行寺幽々子',
    full_name: '西行寺 幽々子（さいぎょうじ ゆゆこ）',
    description: '白玉楼の主。死を司る能力を持つが、おっとりとした性格で食いしん坊。',
    avatar: '/characters/yuyuko.png'
  }
];
